DataFormats/simulation/include/SimulationDataFormat/MCTruthContainer.h

```diff
@@ -43,20 +42,33 @@ struct MCTruthHeaderElement {
 // note that we inherit from TObject just to be able to register this thing with the FairRootManager
 // (which might not be necessary in the future)
 template <typename TruthElement>
-class MCTruthContainer : public TNamed
+class MCTruthContainer
 {
  private:
   std::vector<MCTruthHeaderElement>
     mHeaderArray;                        // the header structure array serves as an index into the actual storage
   std::vector<TruthElement> mTruthArray; // the buffer containing the actual truth information
 
+  int getSize(int dataindex) const
+  {
+    // calculate size / number of labels from a difference in pointed indices
+    const auto size = (dataindex < mHeaderArray.size() - 1)
+                        ? mHeaderArray[dataindex + 1].index - mHeaderArray[dataindex].index
+                        : mTruthArray.size() - mHeaderArray[dataindex].index;
+    return size;
+  }
+
```

Comment:
most probably irrelevant, but what's the type of size? Wouldn't size_t be more appropriate than int?

Reply:
agreed.